src/greynirseq/utils/preprocessing/filters.py

```diff
@@ -0,0 +1,997 @@
+    ice, eng = ex["is"], ex["en"]
+    symbols = re.escape("?")
+    pat = r"\w[" + symbols + "]+\w"
+    prog = RegexCache.compile_rx(pat)
+    found = prog.search(ice) or prog.search(eng)
+    return not found
+
+
+@register_filter
+def improper_line_split(ex):
+    ice, eng = ex["is"], ex["en"]
+    ice_prog = RegexCache.compile_rx(r"(\b(?!\d)\w+)- (\b(?!eða|og)\w+\b)")
+    ice_matches = ice_prog.findall(ice)
+    eng_prog = RegexCache.compile_rx(r"(\b(?!\d)\w+- \b(?!or|and)\w+\b)")
+    eng_matches = eng_prog.findall(eng)
+    return len(ice_matches) == 0 or len(eng_matches) == 0
+
+
+@register_filter
+def dot_pattern(ex):
+    ice, eng = ex["is"], ex["en"]
+    prog = RegexCache.compile_rx("(" + r"\.\s+" + "){2,}")
+    found = prog.search(ice) or prog.search(eng)
+    return not found
+
+
+@register_filter
+def bullet_mismatch(ex):
+    # Suggests misalignment since a bullet is a sentence boundary
+    ice, eng = ex["is"], ex["en"]
+    nice = ice.count("•")
+    neng = eng.count("•")
+    return nice == neng
+
+
+@register_filter
+def max_word_length(ex):
+    ice, eng = ex["is"], ex["en"]
+    word_prog = RegexCache.compile_rx(r"\b\w+\b")
+    ice_words = word_prog.findall(ice)
+    eng_words = word_prog.findall(eng)
+    max_ice_len = 50
+    max_eng_len = 50
+    return (
+        max(len(w) for w in ice_words) <= max_ice_len
+        and max(len(w) for w in eng_words) <= max_eng_len
+    )
+
+
+@register_filter
+def min_word_count(ex):
+    try:
+        isl_toks = [tok for tok in tokenizer.tokenize(ex["is"]) if tok.txt is not None and tok.kind == tokenizer.TOK.WORD]
+        eng_toks = [tok for tok in tokenizer.tokenize(ex["en"]) if tok.txt is not None and tok.kind == tokenizer.TOK.WORD]
+    except TypeError  as e:
+        return True
+    return len(isl_toks) >= DEFAULT_MIN_WORD_COUNT and len(eng_toks) >= DEFAULT_MIN_WORD_COUNT
+
+
+class Gather:
+
+    _store = None
+    _prev_store = None
+    _initialized_with_file = False
+
+    @classmethod
+    def _idemp_init(cls):
+        """ Idempotently initialize, if gather data exists from last run
+            it will be used to filter/transform. Otherwise, pipeline needs
+            to be executed twice. """
+        raise NotImplementedError
+
+    @classmethod
+    def gather(cls, ex):
+        raise NotImplementedError
+
+    @classmethod
+    def save_to_file(cls):
+        # Write gathered data to file
+        # This depends on the data structure the gatherer uses
+        raise NotImplementedError
+
+
+class MinFrequency(Gather):
+
+    auto_pass_len = 3  # for numbers and abbreviations
+    min_freq = 2
+    word_prog = RegexCache.compile_rx(r"\b\w+\b")
+    num_prog = RegexCache.compile_rx(r"^\d+$")
+
+    @classmethod
+    def _idemp_init(cls):
+        if cls._store is not None:
+            return
+        try:
+            os.makedirs(_TMP_DIR, exist_ok=True)
+            path = os.path.join(_TMP_DIR, cls.__name__)
+            with open(path, "r") as fh:
```

Comment:
Recommend setting encoding explicitly to utf-8, i.e. `open(path, "r", encoding="utf-8") `(this is not the default on Windows).

Reply:
Fixed